### DISCUSSION TITLE
Manually update internal form state

### DISCUSSION THREAD

**Discussion by PetarMilisavljevic** [react-hook-form/react-hook-form#10318] [Q&A]
I want to persist the form state without submitting the actual form. I am currently storing the form state in my redux.
```
export interface EditorState {
    activeTabIndex: number | null;
    forms: EditorForm[];
}

export interface EditorForm<T> {
    state: FormState<T>;
}
```
The purpose of this is the possibility to have multiple forms and switching between them. Now, the issue is that i want to persist dirty fields and all "metadata" about the changes. This is why I am saving the form state in redux. 
But the issue i have with this is that i cannot actually update the internal form state. I tried using the function from control as below:
```
    useEffect(() => {
        methods.reset();
        methods.control._updateFormState(form.state);
    }, [tabIndex]);
```
But that does not seem to work. Is there a way I could update the form state manually?

**Comment by bombillazo** [comment]
You could use `rhf-plus` and store certain form state data as metadata

https://github.com/bombillazo/rhf-plus/blob/master/docs/form-metadata.md
